demo/repo/Jenkinsfile

```diff
@@ -22,6 +22,7 @@ stage('QA') {
 milestone 1
 stage('Staging') {
     lock(resource: 'staging-server', inversePrecedence: true) {
+        milestone 2
```

Comment:
> any older build waiting for `lock` is aborted

That is exactly what my patch accomplishes.

> without even trying to acquire the lock

Well, it _was_ waiting to acquire the lock already.

Your variant does not work as well. In that case, after approving build 1, build 3 shows the prompt, but build 2 remains running, even though we can see at this point that it is obsolete. If you approve build 3, it runs to completion, but build 2 proceeds to `input` for reasons TBD (maybe a bug in `LockStepExecution.stop`?), yet even if you approve it, it will abort:

```
[Pipeline] milestone
Trying to pass milestone 1
[Pipeline] stage
[Pipeline] { (Staging)
[Pipeline] lock
Trying to acquire lock on [staging-server]
[staging-server] is locked, waiting...
Superseded by cd/master#3
Lock acquired on [staging-server]
[Pipeline] {
[Pipeline] node
Running on mock-agent-4 in /var/jenkins_home/mock-agents/mock-agent-4/workspace/cd/master
[Pipeline] {
[Pipeline] unstash
[Pipeline] sh
[master] Running shell script
+ cp x.war /tmp/webapps/staging.war
[Pipeline] }
[Pipeline] // node
[Pipeline] input
Does http://localhost:8081/staging/ look good?
Proceed or Abort
Click here to forcibly terminate running steps
[Pipeline] milestone
Trying to pass milestone 2
Canceled since build #3 already got here
[Pipeline] }
Lock released on resource [staging-server]
[Pipeline] // lock
[Pipeline] }
[Pipeline] // stage
[Pipeline] End of Pipeline
Finished: NOT_BUILT
```

You can see that `cancelOldersInSight` has been called, but the `interrupt` did not actually halt the `lock` step (hence the suggestion to escalate to `doTerm`).